### PR TITLE
Add f1 item for global usings

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/68009

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ToddGrun @JoeRobich ptal

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 1]
```diff
@@ -23,568 +23,566 @@
...
+            case SyntaxKind.UsingKeyword when token.Parent is UsingDirectiveSyntax:
+                text = token.GetNextToken().IsKind(SyntaxKind.StaticKeyword)
+                    ? Keyword("using-static")
+                    : Keyword("using");
+                return true;
+            case SyntaxKind.StaticKeyword when token.Parent is UsingDirectiveSyntax:
+                text = Keyword("using-static");
+                return true;
+            case SyntaxKind.GlobalKeyword when token.Parent is UsingDirectiveSyntax:
+                text = Keyword("global-using");
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:445]: This needs a corresponding PR on dotnet/docs I think.

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 2]
```diff
@@ -23,568 +23,566 @@
...
-                    text = Keyword("defaultconstraint");
-                    return true;
-                }
+            case SyntaxKind.UsingKeyword when token.Parent is UsingDirectiveSyntax:
+                text = token.GetNextToken().IsKind(SyntaxKind.StaticKeyword)
+                    ? Keyword("using-static")
+                    : Keyword("using");
+                return true;
+            case SyntaxKind.StaticKeyword when token.Parent is UsingDirectiveSyntax:
+                text = Keyword("using-static");
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:442]: While this is existing code for years, this doesn't actually work and needs a change in dotnet/docs

**Review by Youssef1313 — COMMENTED** [review]
My general feeling, F1 is going into a hard-to-maintain state by having to special case the "context" of each keyword.

I feel like there should be a generic doc page for `global` keyword that explains *all possible contexts of the keyword*. And then Roslyn doesn't need to special case `global` (as alias) vs `global` (in global using) etc. It just redirects to the same page for both, and that page has a brief summary with a link to the long doc page for each.

The same goes for every single keyword where there is too much of special casing here.

This is definitely not something to address for this PR, but wanted to comment my feelings for F1 :)

Note that my suggestion above, if you decided to go with it, will require coordination with dotnet/docs team.

I'm starting to think more about this as a docs issue. There is already this doc page with links to all keywords:

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/

I find that this may be confusing to beginners. If you open `global` from that page, it redirects to page about `::` (https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/namespace-alias-qualifier).

To me, it should redirect to a page that has links to both `::` and `global using`.

Then, for any `global`, Roslyn just redirects to the same page without any special casing.

cc @BillWagner as well for thoughts on organizing the docs that way.

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:445]: yes.  that will come after this.

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:442]: there is an issuue on .net docs on this already.

**Comment by CyrusNajmabadi** [comment]
@Youssef1313 i think this is complimentary.  Docs can update how they want, and we can provide the right context for it to take the user to the best location.  Worst case is that roslyn provides more contxt, while docs just redirects to one common location.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**